### PR TITLE
Style fixes from py3 branch

### PR DESCRIPTION
**Pull request by anandtrex**
Addresses #8 

Style fixes, adding style script. 

NOTE: While the run-style-check.sh script has been added, all the issues it points to have not been address. The issues that weren't fixed by autopep8 require manual intervention and cannot be automated.

### CODE REVIEW

File: utils/run-style-check.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+flake8 --ignore=E501,W293,W291,E265,E231,E127,E262,E241 --max-line-length=120 --exclude style-reports/ --tee --output-file style-reports/report-${now}.txt
+# echo $?
+# E501 line too long (89 > 79 characters)
+# W293 blank line contains whitespace
+# W291 trailing whitespace
+# E265 block comment should start with '# '
+# E231 missing whitespace after ':'
+# E127 continuation line over-indented for visual indent
+# E262 inline comment should start with '# '
+# E241 multiple spaces after ','
```
Comment (mhoff) [inline utils/run-style-check.sh]: What exactly are your reasons for those ignored conventions?

For instance, why not W291?

**Comment by kappeld** [comment]
Thanks @anandtrex! Looks great to me!

**Comment by anandtrex** [comment]
Well they are somewhat arbitrary -- a combination of some commonly used
styles e.g. E241 is ignored because spore uses visually aligned formatting
for some list, and things that are difficult to automate or fix manually.

I added W291 for some good reason for LTL (this script is copied over from
the LTL package) but can't remember what the reason was.

I can remove it and others if you want, and make the exceptions spore
specific.

On 25 Apr 2017 07:39, "Michael Hoff" <notifications@github.com> wrote:

*@mhoff* commented on this pull request.
------------------------------

In utils/run-style-check.sh
<https://github.com/IGITUGraz/spore-nest-module/pull/11#discussion_r113110828>
:

> @@ -0,0 +1,13 @@
+mkdir style-reports
+now=$(date +"%m-%d-%Y-%T")
+flake8 --ignore=E501,W293,W291,E265,E231,E127,E262,E241
--max-line-length=120 --exclude style-reports/ --tee --output-file
style-reports/report-${now}.txt
+# echo $?
+# E501 line too long (89 > 79 characters)
+# W293 blank line contains whitespace
+# W291 trailing whitespace
+# E265 block comment should start with '# '
+# E231 missing whitespace after ':'
+# E127 continuation line over-indented for visual indent
+# E262 inline comment should start with '# '
+# E241 multiple spaces after ','

What exactly are your reasons for those ignored conventions?

For instance, why not W291?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/IGITUGraz/spore-nest-module/pull/11#pullrequestreview-34472831>,
or mute the thread

**Comment by mhoff** [comment]
Most of these ignored rules are not necessary for our project. Ignoring (only) `E241,F403,F405` yields:

```
./examples/pattern_matching_showcase/experiment.py:32:4: E713 test for membership should be 'not in'
./examples/pattern_matching_showcase/python/__init__.py:49:121: E501 line too long (121 > 120 characters)
./examples/pattern_matching_showcase/python/zmq_proxy_node.py:4:1: F401 'time' imported but unused
./examples/pattern_matching_showcase/python/config.py:26:1: F401 'os' imported but unused
./examples/pattern_matching_showcase/python/config.py:27:1: F401 'math' imported but unused
./examples/pattern_matching_showcase/python/config.py:29:1: F401 'time' imported but unused
./examples/pattern_matching_showcase/python/control_node.py:8:1: F401 'itertools.takewhile' imported but unused
./examples/pattern_matching_showcase/python/control_node.py:8:1: F401 'itertools.dropwhile' imported but unused
./examples/pattern_matching_showcase/python/control_node.py:95:37: F841 local variable 'e' is assigned to but never used
./examples/pattern_matching_showcase/python/interface.py:49:121: E501 line too long (121 > 120 characters)
./examples/pattern_matching_showcase/python/network_node.py:8:1: F401 'numpy as np' imported but unused
./examples/pattern_matching_showcase/python/snn_utils/plotter/data_provider.py:52:13: E265 block comment should start with '# '
./examples/pattern_matching_showcase/python/snn_utils/comm/music/__init__.py:149:20: F402 import 'buffer' from line 6 shadowed by loop variable
./examples/pattern_matching_showcase/python/snn_utils/comm/music/__init__.py:168:13: F402 import 'buffer' from line 6 shadowed by loop variable
./examples/pattern_matching_showcase/python/snn_utils/comm/music/__init__.py:174:13: F402 import 'buffer' from line 6 shadowed by loop variable
./examples/pattern_matching_showcase/python/snn_utils/comm/zmq/__init__.py:59:13: E704 multiple statements on one line (def)
./examples/pattern_matching_showcase/python/snn_utils/comm/zmq/__init__.py:61:13: E704 multiple statements on one line (def)
./examples/pattern_matching_showcase/python/snn_utils/comm/zmq/__init__.py:64:13: E704 multiple statements on one line (def)
./examples/pattern_matching_showcase/python/snn_utils/comm/zmq/__init__.py:67:17: E704 multiple statements on one line (def)
./examples/pattern_matching_showcase/python/snn_utils/comm/zmq/__init__.py:70:17: E704 multiple statements on one line (def)
./unittests/test_dbl_exp_neuron.py:56:121: E501 line too long (156 > 120 characters)
./unittests/test_dbl_exp_neuron.py:90:121: E501 line too long (189 > 120 characters)
./unittests/test_garbage_collector.py:106:9: F841 local variable 'p_sim' is assigned to but never used
./unittests/test_garbage_collector.py:109:9: F841 local variable 'synapse_properties' is assigned to but never used
./unittests/test_garbage_collector.py:115:9: F841 local variable 'target_values' is assigned to but never used
./unittests/test_dbl_exp_neuron_rate.py:42:121: E501 line too long (156 > 120 characters)
```
```
# E241 multiple spaces after ','
# F403 star imports
# F405 undefined or from star import
```

I added `F403` and `F405` because I make heavy use of star-imports with my global configuration file. I agree that `E241` makes sense for alignment, pycharm however did already break this nice alignment for many files I think (we naively used auto-format).

I would be fine with sticking to these three ignores for now.

**Comment by anandtrex** [comment]
I would suggest including E501 also since in many places readability
becomes worse if reformatted to 120 chars.

Ideally we should decide the list to ignore when someone is actually trying
to fix the warnings. Otherwise it's mostly speculation :-)

**Comment by mhoff** [comment]
True. So how do we proceed? Add E501 to the list and merge?

**Comment by anandtrex** [comment]
I would suggest merge this in, and have a separate issue for fixing style errors, which can include adjusting errors we want to ignore in `run-style-check.sh`.

**Comment by mhoff** [comment]
Like this?

**Comment by anandtrex** [comment]
I've created #12 to address this. I'm going to merge this in.

**Pull request merged**